app/controllers/api/v1/publications_controller.rb

```diff
@@ -253,5 +337,65 @@ def filter_by_doi(query)
 
         query.where(doi: doi)
       end
+
+      def params_valid?
+        ai_id = params[:activity_insight_id]
+        doi = params[:doi]
+
+        (ai_id.present? && doi.nil?) ||
+          (ai_id.nil? && doi.present?) &&
+            params[:scholarsphere_open_access_url].present?
```

Comment:
if both ai_id and doi are present, it's false, so we could rewrite it:

``` ruby
def params_valid?
  return false unless params[:scholarsphere_open_access_url].present?

  !(params[:activity_insight_id].present? && params[:doi].present?)
end
```

Reply:
yes 👌🏻 ☝🏻